lib/waves.rb

```diff
@@ -0,0 +1,57 @@
+class Waves
+  def initialize
+    @iteration = 1
+    @wave1_image = Gosu::Image.new('media/wave1.png')
+    @wave2_image = Gosu::Image.new('media/wave2.png')
+    @wave3_image = Gosu::Image.new('media/wave3.png')
+
+  end
+
+  # apply velocity to the position
+  def update
+    @iteration += 1
```

Comment:
use delta-time instead `def update(dt)`

Reply:
Can you explain this more? Do you want me to make this update dependent on the time that has passed? dt and delta-time are the time step, correct?

Reply:
if you look in https://github.com/laurcor55/sailor-game/pull/10/files#diff-81aed5f475932a4e2caeedccb9c37748R14 we are using `dt` to tell update methods the number of milliseconds since the last time `Window#update` was called. it's not always going to be the same for various reasons, but you can get a pretty close approximation to where an object should be if you have `dt`. like if your machine slows down you should still be able to predict (with better accuracy) where the wave is just based on time elapsed

Reply:
in other words, if you dont use delta-time, the waves will move faster/slower on different machines

Reply:
oh that makes sense. I have always seen dt and delta-time used as time step. Can we maybe call dt "elapsed-time" or something?

Reply:
`dt` is pretty standard as far as i know. it means a specific thing in a game loop. it means "change in time since last iteration". the only difference than what you're thinking is that here `dt` is not static.

Reply:
but yeah if your reeaaalllyy want to you can use "elapsed time", but make sure you update the rest of the code as well to keep it all consistent

Reply:
wait so delta-time or dt is the overall elapsed time?